pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -395,22 +395,22 @@ stdenv.mkDerivation (finalAttrs: {
       --replace /usr/local/lib/frei0r-1 ${frei0r}/lib/frei0r-1
   '';
 
-  patches = map (patch: fetchpatch2 patch) ([ ]
```

Comment:
```
  patches = [
    ...
  ] ++ map (patch: fetchpatch2 patch) ([ ]
```

Wouldn't this work?

Reply:
But then I'd need to repeat the whole `optionals (lib.versionAtLeast version "5" && lib.versionOlder version "6")` stuff for the non-fetchpatched patches

Reply:
You could do a hack like

```Nix
map (patch: if builtins.isPath patch then patch else fetchpatch2 patch)
```

but I don't have a strong opinion on this.

Reply:
I just don't think it's a hugely useful piece of DRY in the first place, don't think we need the added complexity.